Wald's Equation and Renewal Theory

** note this isn't actually from Feller but instead from page 422 of Ross's Intro To Probability Models and ties in with some exercises in Gallagher**  

consider applying Wald's Equality where 

$0 \lt X$  

(recall that $X(\omega)$ is a time denominated increment)  and $X_i$ are iid

$\bar{ X} \lt \infty$  

and $S_n = X_1 + X_2 +... + X_n$  

and with the stopping rule of stopping at the first cumulative arrival time $\gt t$ 

**open item: considering doing this for $\geq t$, the stopping rule is intact... the issue really seems to be notational** 

that is, we are considering 

$S_{N(t)+1}$  

where $N(t)$ is the number of arrivals at time $t$ 

$S_{N(t)+1} = t  + Y(t)$  

where $Y(t)$ is the excess at time $t$, i..e. the final $X_i$ arrial is partitioned into the amount up to and including $t$ and the amount after -- the latter is what's counted by $Y(t)$. For avoidance of doubt $Y(t)\geq 0$ 

taking expectations of both sides, we have 


$E\big[S_{N(t)+1}\big] = E\big[t  + Y(t)\big] = E\big[X_1 + X_2 +... + X_{N(t)} + X_{N(t)+1}\big]$  

$t + E\big[ Y(t)\big] = \bar{X}E\big[{N(t)+1}\big] = \bar{X}E\big[{N(t)}\big]+ \bar{X}$  

$E\big[{N(t)}\big]+ 1 = \frac{t}{\bar{X}} + \frac{E\big[ Y(t)\big]}{\bar{X}}$  

$E\big[{N(t)}\big] = \frac{t}{\bar{X}} + \frac{E\big[ Y(t)\big]}{\bar{X}} - 1$  


$\frac{t}{\bar{X}}  - 1 \leq E\big[{N(t)}\big] = \frac{t}{\bar{X}} + \frac{E\big[ Y(t)\big]}{\bar{X}} - 1 = \frac{t}{\bar{X}} - 1 + \frac{E\big[ Y(t)\big]}{\bar{X}} $  

sometimes the notation $m(t) :=  E\big[{N(t)}\big]$ is used instead.  This is then re-written as 

$\frac{t}{\bar{X}}  - 1 \leq m(t) = \frac{t}{\bar{X}}  - 1 + \frac{E\big[ Y(t)\big]}{\bar{X}}$   

**some important special cases** 

*infinite expected waiting time*
we know that it may be possible for expected waiting times to be infinite, if the renewal does not have a second moment.  In such a case, by positivity, we can still be assured that the lower bound is valid.  As for the upper bound, we'd re-run the above argument, but on a truncated $\hat{X}$, and take limits as needed. Carefully pursuing this (as in Ross, Gallagher, or Feller) then recovers the elementary renewal theorem.  
- - - -
Of more interest for this writeup are caes where $E\big[ Y(t)\big]$ is finite *and* we may estimate it or find a useful upper bound for it. This motivates cases of  


*monotonocity in a graph *  

note: for *many* special cases of interest we may be about to bound $E\big[ Y(t)\big]$ 

In a particular special case of interest -- when there is *monotonicity* in the graph / underlying countable state markov chain, we **know** that any possible renewal from the starting node (call it 0)  must go through all other nodes, which gives a dominance relation of 

$\{\text{time from 0 }  \to \text{ k}\} + \{\text{time from k }  \to \text{0}\} = \{\text{time from 0 }  \to \text{ 0}\} $  

for $k \gt 0$  


over any legal sample path, and for avoidance of doubt, both quantities being time denominated are real non-negative.  

taking expectations we have 

$E\big[\text{time from 0 }  \to \text{ k} \big] + E\big[\text{time from k}  \to \text{ 0}\big] = E\big[\text{time from 0 }  \to \text{ 0}\big] = \bar{X} $  

$E\big[\text{time from 0 }  \to \text{ k} \big] \leq \bar{X} $  

Thus we can write 

$E\big[ Y(t)\big] = \sum_{k\gt 0} p_k E\big[\text{time from 0 }  \to \text{ k} \big] \leq \sum_{k\gt 0} p_k \bar{X} = \bar{X} \sum_{k\gt 0} p_k \leq \bar{X} $  

where $p_k$ denotes the probability of being in state $k$ at the stopping time.  By definition we have 

$0 \leq \sum_{k\gt 0} p_k \leq  p_0 + \sum_{k\gt 0} p_k = \sum_{k\geq 0} p_k  = 1$  

- - - - 

In these cases of monotonicity, we may then say:  


$\frac{t}{\bar{X}}  - 1 \leq m(t) = \frac{t}{\bar{X}} -1 + \frac{E\big[ Y(t)\big]}{\bar{X}} \leq \frac{t}{\bar{X}}-1 + \frac{\bar{X}}{\bar{X}}  = \frac{t}{\bar{X}} -1 +1 = \frac{t}{\bar{X}}$   

So we may conclude that 

$\frac{t}{\bar{X}}  - 1 \leq m(t) \leq \frac{t}{\bar{X}}$    



**Alternating Renewal Theorem:  **  

Suppose that we have iid $X_n$ and iid $Y_n$ where $X_n$ denotes "on" time and $Y_n$ is off time. Each random variable denotes an arrival time, and they come in sequence -- i.e. first always $X_i$ then $Y_i$ then $X_{i+1}$ then $Y_{i+1}$ then $X_{i+2}$ and so on.  Further suppose that each random variable has a first moment.  

Thus $Z_n := X_n + Y_n$  is a renewal process that starts with the system off, 'turns on' with the arrival denoted by $X$ and then turns off again at the $Y$ arrival -- which constitutes a renewal.  

An interesting result: 

$\lim_{t \to \infty} P\big[\text{system is "on" at time t}\big]= \frac{\bar{X}}{\bar{X} + \bar{Y}} = \frac{E\big[X_1\big]}{E\big[X_1 + Y_1\big] }$  


we suggest two different proofs / ways of deriving this result.  One using the Renewal Equilibrium theorem (variously called Feller-Erdos-Pollards for arithmetic/lattice cases and Blackwell's theorem for non-arithmetic case), and the other approach is to find that the above conforms to the time averaged result -- hence if there is a limmitting distribution / equilibrium for $Z$ then the above relationship must follow.  (As always the time averaged case has an interpretation in terms problems of 'random incidence' for a finite but sufficiently large $t$.)  


**1.)** 

Approach one: using the key limit theorem, we know the CDF of the age (**2x check on residual life**) is given by 

$\lim_{t \to \infty} Z_{(t)}\big(z\big) =\frac{Pr\{Z \gt (z)\} }{E\big[Z_1\big]} = \frac{Pr\{Z \gt (z)\} }{E\big[X_1 + Y_1\big]} = \frac{Pr\{Z \gt (z)\} }{E\big[X_1\big] + E\big[Y_1\big]}$  

of course if we integrate over $z$ from 0 to $\infty$ we get a 

$\frac{\int_0^{\infty} Pr\{Z \gt (z)\}dz }{E\big[X_1\big] + E\big[Y_1\big]} = \frac{E\big[Z_1\big] }{E\big[X_1\big] + E\big[Y_1\big]} =1 $

which confirms that this is a valid probability distribution.  

However, we can further refine this.  For convenience note that 

$E\big[\mathbb I_{Z_i \gt z}\big] = E\big[\mathbb I_{Z \gt z}\big] = Pr\{Z \gt (z)\} $  

The indicator on the left can be further decomposed, taking advantage of the sequential ordering of arrivals.  

$\mathbb I_{Z_i \gt z} = \mathbb I_{X_i \gt z} + \mathbb I_{\text{convolution :} X\leq z * Y \gt (z-x)}$  

taking expectations we have 

$ Pr\{Z \gt (z)\} = E\big[\mathbb I_{Z_i \gt z}\big] = E\big[\mathbb I_{X_i \gt z}\big] + E\big[\mathbb I_{\text{convolution :} X\leq z * Y \gt (z-x)}\big] =  Pr\{X \gt (z)\}  + E\big[\mathbb I_{\text{convolution :} X\leq z * Y \gt (z-x)}\big]$  

Now if we only wanted the portion of time that is "on", that is given directly by $Pr\{X \gt (z)\}$ -- and if integrate over this, we'd have 'summed' over all probabilities of being on in this equilibrium state, which is given below as:  


$\frac{\int_0^{\infty} Pr\{X \gt (z)\} dz}{E\big[X_1\big] + E\big[Y_1\big]} = \frac{E\big[X_1\big] }{E\big[X_1\big] + E\big[Y_1\big]} = \frac{\bar{X}}{\bar{X} + \bar{Y}}$  

as desired 


**2.)** 

Time averaged approach: As before, we assume each $X_i$ and $Y_i$ have a first moment.  Further to cut to the chase for this argument, assume the random variables have a second moment (or if desired: are  bounded-- which easily sets this up for a truncation argument if needed for full generality)

Use the argument at the top of this workbook with a stopping trial that stops at the first $Z$ arrival after some time $t$ and apply Wald's Equality.  However, note that we are modelling a slightly different result, where we have:  

$S_n := Z_1 + Z_2 +... + Z_n$  

$E\big[S_{N(t)+1}\big] = E\big[t  + R_L(t)\big] = E\big[Z_1 + Z_2 +... + Z_{N(t)} + Z_{N(t)+1}\big] = \bar{Z}\big(N(t)\big) = \bar{Z}\cdot m(t)$ 

where $R_L(t)$ denotes the residual life as a function of time (formerly this was $Y(t)$ though we have a notational overload.)  

if we further re-run the argument as before, we get:  

$\frac{t}{\bar{Z}}  - 1 \leq m(t) = \frac{t}{\bar{Z}}  - 1 + \frac{E\big[ R_L(t)\big]}{\bar{Z}}$   

dividing by $t$  
$\frac{1}{\bar{Z}}  - \frac{1}{t} \leq \frac{m(t)}{t} = \frac{1}{\bar{Z}}  - \frac{1}{t} + \frac{E\big[ R_L(t)\big]}{\bar{Z}\cdot t} \leq \frac{1}{\bar{Z}} + \frac{E\big[ R_L(t)\big]}{\bar{Z}\cdot t}$   

taking advantage of our assumption (for convenience) that $E\big[ R_L(t)\big]$ is bounded we look at the limitting value as $t \to \infty$ and see 

$\frac{1}{\bar{Z}} \leq \frac{m(t)}{t} \leq \frac{1}{\bar{Z} }  $   

as expected.  

*However* we are actually interested in estimating $p$ which is the long-run portion of time that the system is "on".  So we set up a stopping trial 

$p(t)\cdot S_{N(t)+1} = X_1 + X_2 + ... + X_{N(t)}+ X_{N(t)+1}$ 

where $p(t) := \frac{X_1 + X_2 + ... + X_{N(t)}+ X_{N(t)+1}}{Z_1 + Z_2 + ... + Z_{N(t)}+ Z_{N(t)+1}}$   

for avoidance of doubt: the trials stop on arrival $Z_{N(t)+1}$ -- so there cannot be a 'mismatch' between the $X$'s and $Z$'s -- we always stop on the first renewal after time $t$.  

in words: $p(t)$ is distribution of the portion of time that the system is 'on', as a function of time.   It can be a touch confusing since it is a random variable that reflects proportions / probabilities for other random variables.  But ultimately $p(t)$ is just another a random variable taking on values $\in [0,1]$ -- and it has a CDF which gives the probability of $p(t)$ being less than $x$ for some chosen time $t$. Note that this means for some time $t$ it also has a complementary CDF of $Pr\{p(t) \gt x\}$ and if we integrate over this, we'd recover the expected portion amount of time that the system is on at some time $t$.  While these CDFs exist, they are not easy to tackle head on.  The same holds for the expected value of $p(t)$.  Instead we will be estimating it via setting up a stopping trial and applying Wald's Equality, and with an ultimate goal of passing a limit.  

Ultimately we know want to find the probability as $t \to \infty$ where we simply show $p = p(\infty) $.  Technically we are looking at the expected value of this limmitting probability 

-- as a purely technical point note:  like any probability, $p(t)$ is bounded in $[0,1]$  so if needed we may use bounded convergence theorem here to justify that 

$p = E\big[\lim_{t \to \infty}p(t)\big] = \lim_{t \to \infty}E\big[p(t)\big]$

- - - - 

Now we re-run our argument and see:  


$E\big[p(t)\cdot S_{N(t)+1}\big] = E\big[p(t)\cdot \big(t  + R_L(t)\big)\big]= E\big[p(t)\cdot t  + p(t) \cdot R_L(t)\big] = E\big[X_1 + X_2 +... + X_{N(t)} + X_{N(t)+1}\big] = \bar{X}\cdot E\big[N(t) +1 \big] $ 

$= t\cdot E\big[p(t)\big] + E\big[p(t) \cdot R_L(t)\big] =   \bar{X}\cdot m(t) +\bar{X}$  

re-arranging terms:   
$t\cdot E\big[p(t)\big] + E\big[p(t) \cdot R_L(t)\big]- \bar{X} =  \bar{X}\cdot m(t) $  

dividing by $\bar{X}$  


$t \cdot \frac{ E\big[p(t)\big]}{\bar{X}}  - 1  \leq t \cdot \frac{ E\big[p(t)\big]}{\bar{X}} + \frac{E\big[p(t) \cdot R_L(t)\big]}{\bar{X}} -1 =   m(t) \leq t \cdot \frac{ E\big[p(t)\big]}{\bar{X}} + \frac{E\big[p(t) \cdot R_L(t)\big]}{\bar{X}}\leq t \cdot \frac{ E\big[p(t)\big]}{\bar{X}} + \frac{E\big[ R_L(t)\big]}{\bar{X}}   $  

recalling that $p(t)$ is bounded between 0 and 1 and $R_L(t)$ is real non-negative 

dividing by $t$ we have 

$ \frac{ E\big[p(t)\big]}{ \bar{X}}  - \frac{1}{t}  \leq  \frac{ m(t)}{t}  \leq   \frac{ E\big[p(t)\big]}{\bar{X}} + \frac{ E\big[R_L(t)\big]}{\bar{t \cdot X}}   $ 

passing limits as before, and taking advantage of boundedness we see as $t \to \infty$  

$ \frac{ E\big[p(t)\big]}{ \bar{X}} = \frac{ p}{ \bar{X}}   \leq  \frac{ m(t)}{t} = \frac{1}{\bar{Z} }  \leq   \frac{ E\big[p(t)\big]}{\bar{X}} =\frac{ p}{\bar{X}}  $ 

Where we made the subsitution in the middle from the argument that began this second approach. 
- - - -  

We thus see that in the limit, we have a condition where:  

$\frac{1}{\bar{Z} }  =  \frac{ p}{\bar{X}} $

or 

$p = \frac{\bar{X}}{\bar{Z} }  =   \frac{\bar{X}}{\bar{X} + \bar{Y} } $  

as desired  

**follow-up: not just the limitting value-- I want to show that residual life in general must be greater than the expected renewal rate... some inequalities on this are in 2 of Ross's texts** 


maybe set this up as a point-wise bound using Indicator random variables... 



In [89]:
import numpy as np
# python 3.x

m = 16
p = 0.4

A = np.zeros((m,m))
B = np.zeros((m,m))

A[0,-1]= 1-p
A[1,-1]= p

B[0,-1]= 1
##
A[0,0] = 1-p
A[1,0] = p
B[0,0] = 1-p
B[1,0] = p


for k in range(0, m-1):
    A[k+1,k] = p
    A[0,k] = 1-p
 
    B[k+1,k] = p
    B[0,k] = 1-p

print(A,"\n")
print(B)


[[ 0.6  0.6  0.6  0.6  0.6  0.6  0.6  0.6  0.6  0.6  0.6  0.6  0.6  0.6
   0.6  0.6]
 [ 0.4  0.   0.   0.   0.   0.   0.   0.   0.   0.   0.   0.   0.   0.   0.
   0.4]
 [ 0.   0.4  0.   0.   0.   0.   0.   0.   0.   0.   0.   0.   0.   0.   0.
   0. ]
 [ 0.   0.   0.4  0.   0.   0.   0.   0.   0.   0.   0.   0.   0.   0.   0.
   0. ]
 [ 0.   0.   0.   0.4  0.   0.   0.   0.   0.   0.   0.   0.   0.   0.   0.
   0. ]
 [ 0.   0.   0.   0.   0.4  0.   0.   0.   0.   0.   0.   0.   0.   0.   0.
   0. ]
 [ 0.   0.   0.   0.   0.   0.4  0.   0.   0.   0.   0.   0.   0.   0.   0.
   0. ]
 [ 0.   0.   0.   0.   0.   0.   0.4  0.   0.   0.   0.   0.   0.   0.   0.
   0. ]
 [ 0.   0.   0.   0.   0.   0.   0.   0.4  0.   0.   0.   0.   0.   0.   0.
   0. ]
 [ 0.   0.   0.   0.   0.   0.   0.   0.   0.4  0.   0.   0.   0.   0.   0.
   0. ]
 [ 0.   0.   0.   0.   0.   0.   0.   0.   0.   0.4  0.   0.   0.   0.   0.
   0. ]
 [ 0.   0.   0.   0.   0.   0.   0.   0.   0.   0.   0.4  0.   0.   0.   0.

In [90]:
t_trials = 5000000

x = np.zeros(A.shape[0])
x[0] = 1

onesv = np.ones(A.shape[0])
counter = 0 

for _ in range(t_trials):
    b = A @ x
    counter += b[-1] 
    x = b
#     print(b)
    
counter

3.2212203404772941

In [91]:
1/(counter/t_trials)

1552206.7637444327

In [92]:
Xbar = 0
for k in range(1, m):
    Xbar += 1/p**k
Xbar

1552202.6243591295

In [93]:
(1-p**(m-1))/((1-p)*p**(m-1))

1552202.6243591297

In [94]:
t_trials/Xbar

3.2212289307682305

More interesting cases of Monotonicity in Graphs 

Some bounds on second largest eigenvalue of an absorbing state markov chain 

referencing the two different forms of Collatz-Wielandt formula for Positive matrices (on p 667 - 669 of Meyers *Matrix Analysis*)  






In [3]:
T, Z = scipy.linalg.schur(A)
T

NameError: name 'scipy' is not defined

In [ ]:
np.conj(Z).T

In [ ]:
1/ 0.25

In [ ]:
value = 0.250184
# value = 1/4
running_sum = 0
for i in range(1, 5+1):
    running_sum += i*3*value**i
running_sum

In [ ]:
1/(1-value)

In [ ]:
0.25/0.250184

# the below is really more in line with the Markov Chains and Matrix Theory but still closely tied in with the above

In [ ]:
16*3

In [ ]:
theta = 1/2

x = np.array([theta, theta**2, theta**3, theta**4, theta**5, theta**6, 2*theta**7])
x

In [ ]:
counter = np.arange(x.shape[0]) + 1


In [ ]:
counter@ x

In [ ]:
# nfl scores problem
# 2, 3, 6, 7, 8
# 1, 4, 5 are not legal

C = np.zeros((8,8))
for k in range(8-1):
    C[k+1, k] = 1
C[0] += 1

C[0,0] = 0
C[0,4] = 0
C[0,3] = 0

C

print(C)

x = np.zeros(8)

x[8-2] = 1
# score of 2
x[8-3] = 1
# score of 3
x[8-6] = 3
# score of 6
x[8-7] = 2
# score of 7 
x[8-8] = 4
# score of 8
print(x,"\n")

for _ in range(30):
    b = C @ x
    print(x@ b / np.linalg.norm(x)) 
    print(b," \n")
    x = b
#     x = x/np.linalg.norm(x)



In [ ]:
np.linalg.eigvals(C)

In [ ]:
1/0.68233

In [ ]:
C = np.zeros((5,5))
for k in range(5-1):
    C[k+1, k] = 1
C[0] += 3
C
eigs = np.linalg.eigvals(C)

# C *= 1/abs(eigs[0])
C

In [ ]:
np.linalg.matrix_power(C, 1000)

In [ ]:
# x = np.random.random(5)
# x = x / np.linalg.norm(x)
x = np.zeros(5)
x[0] = 1
t_iterations = 100

for _ in range(t_iterations):
    b = C @ x
#     print( np.linalg.norm(b)/np.linalg.norm(x)  )
#     x = b/ np.linalg.norm(x)
    x = b
    

In [ ]:
x

In [ ]:
np.linalg.norm(x)

the idea then is we want to select some
$\delta \gt 0$ 

where for any $\theta \in (0,1)$ we still have $0 \lt \theta + \delta \lt 1$  

such that 

$\frac{1 - (\theta + \delta)^6}{1-(\delta +\theta)} \gt \frac{1}{1-\theta}$  



In [ ]:
theta = (- 1 + np.sqrt(5))/ 2

C = np.array([[theta, theta**2],[1,0]])
C

In [ ]:
mu = theta + 2*theta**2

In [ ]:
1/mu

In [ ]:
np.linalg.matrix_power(C, 100)

# Problem 12

# Problem 19

note: this comes up again on page 443, XVI -- for a markov chain application

On infinite renewals for non-defective renewal processes  

see: my notes under problem 2.28 in chp2 notes to Gallagher's Stochastic Processes 



**comment:**  The problem's result is ultimately clear.  However it feels almost like a trick amongst identities of Generating Functions.  The problem is contained in the 'answers' section in the back, though it took your author a considerable amount of time to parse those answers and recover the missing steps to make things coherent.  



**Key Results:**   

$U(s) = \frac{1}{1 - F(s)} = \sum_{k=0}^\infty u_k s^k$  
this is (3.2) on page 311

recall per page 308, that $\{u_k\}$ are the probability of there being renewal at $k$ -- for non-transient setups, this sums to $\infty$ so cannot be a probability.  but the series converges for $\vert s\vert \lt 1$ (recall chapter XI).  

The answers in the back note follwing identities

(a) $1-F(s) = (1-s) Q(s)$ (to be used first)   
(b)$\bar{X} - Q(s) = (1-s)R(s)$ (to be used a bit later)    

(note I chose to use $\bar{X}$ not $\mu$ here to avoid visual confusiton with the $u_k$'s)  

**Goal: **

assuming we have a second moment, show 

(1.) $\sum_{k=0}^\infty \big(u_k - \frac{1}{\bar{X}}\big)s^k = \frac{R(s)}{\bar{X}Q(s)}$  

for any $s \in (0,1)$ 

(2.) then show it in fact holds for $s:=1$ as well, with no convergence issues.  and at s = 1 
(or as I prefer, taking the limit as $s$ approach $1$ from the left)  


$\sum_{k=0}^\infty \big(u_k - \frac{1}{\bar{X}}\big)= \frac{\sigma^2 - \bar{X} + \bar{X}^2}{2 \bar{X}^2}$ $= \frac{E\big[X^2\big] - \bar{X}}{2\bar{X}^2}$  

note: the above is interpretted as the total lifetime difference between the probability of a renewal at a particular time, and the steady state probability of renewal at that time 

**Proof:**  

Starting with (1.), we build the relation we want starting with $s\in(0,1)$ where we have absolute convergence 

$\sum_{k=0}^\infty \big(u_k - \frac{1}{\bar{X}}\big)s^k = \big(\sum_{k=0}^\infty u_k s^k\big) -  \big(\sum_{k=0}^\infty \bar{X}^{-1}s^k \big) = \big(U(s)\big) -\big(\frac{\bar{X}^{-1}}{1-s} \big) = \big(\frac{1}{1 - F(s)}\big) -\frac{\bar{X}^{-1}}{1-s} = \frac{1}{(1-s) Q(s)} -\frac{1}{(1-s)\bar{X}}$ 

where we used identity (a) on the Right Hand Side 






$\sum_{k=0}^\infty \big(u_k - \frac{1}{\mu}\big)s^k = \frac{1}{(1-s) Q(s)} -\frac{1}{(1-s)\bar{X}} = \frac{\bar{X}}{(1-s) \bar{X}Q(s)} -\frac{Q(s)}{(1-s)\bar{X}Q(s)}= \frac{\bar{X} - Q(s)}{(1-s)\bar{X}Q(s)} = \frac{(1-s)R(s)}{(1-s)\bar{X}Q(s)}= \frac{R(s)}{\bar{X}Q(s)}$ 

where we used identity (b), on the Right hand side, and 'divided out' $(1-s)$, an operation which is valid for any $s \neq 1$  

(Note: official solution uses this reduced form to make claims directly about the LHS and RHS at $s =1$... the legacy of the $(1-s)$ that *was* there in my derivation makes me a touch concerned, and instead I prefer to show the relation is true for $1 - \delta$ small positive $\delta$.  There might be a slightly simpler or more clean approach to get to these results. As a related point, I have lingering concerns about how I explicitly split the geometric series above -- this is fine as we have absolute convergence for $s = 1 - \delta$, but techincally that intermediate step doesn't stand on its own if we re-examine it at $s=1$.)  




This completes part (1.)


- - - -
part (2.)  

we recall that $Q(1) = \bar{X}$  (i.e. summing / integrating over the complementary CDF) thus we also know 

$\lim_{s \to 1^{-}}Q(s) = Q(1) = \bar{X}$  

(i.e. approach one as a limit from the left)  

and we also recall that 

$R(1) = \frac{\sigma^2 - \bar{X} + \bar{X}^2}{2} = \frac{E\big[X^2\big] - \bar{X}}{2}$   

since we have a second moment by assumption, this exists and is finite 

thus

$\lim_{s \to 1^{-}}R(s) = \frac{\sigma^2 - \bar{X} + \bar{X}^2}{2} = \frac{E\big[X^2\big] - \bar{X}}{2}$  

This is discussed and shown in multiple ways in my Absorbing State Markov Chains writeup.  It is also discussed and shown directly on page 266 (i.e. beginning of XI, discussing generating functions).  

Finally 

$\lim_{s \to 1^{-}} \sum_{k=0}^\infty \big(u_k - \frac{1}{\mu}\big)s^k =  \lim_{s \to 1^{-}}\frac{R(s)}{\bar{X}Q(s)} =  \frac{\sigma^2 - \bar{X} + \bar{X}^2}{2\bar{X}^2} =  \frac{E[X^2] - \bar{X}}{2\bar{X}^2}$   

note: the book makes a slightly stronger claim at the end and states

$\sum_{k=0}^\infty \big(u_k - \frac{1}{\mu}\big) = RHS$  

i.e. it directly evaluates the LHS at $s=1$.  There are some analytic subtleties in the problem and how I approached it that I don't think I fully grasp, hence I am content to evaluate instead approaching the limit from the left.  

